pkg/awsutils/awsutils.go

```diff
@@ -187,6 +189,14 @@ func (cache *EC2InstanceMetadataCache) initWithEC2Metadata() error {
 	}
 	log.Debugf("Found instance-id: %s ", cache.instanceID)
 
+	// retrieve instance-type
+	cache.instanceType, err = cache.ec2Metadata.GetMetadata(metadataInstanceType)
+	if err != nil {
+		log.Errorf("Failed to retrieve instance-type from instance metadata %v", err)
+		return errors.Wrap(err, "get instance metadata: failed to retrieve instance-type")
```

Comment:
L-IPAM will NOT functions properly if it can not retrieve instance metadata.  I think it is better to fail fast, so that APIServer/Scheduler/Controller will NOT schedule any pods onto this problematic node.